lib/chooseUserGrid.js

```diff
@@ -0,0 +1,68 @@
+const rl = require("./rl");
+
+const promptForUserGridChoice = () => {
+  return new Promise((resolve) => {
+    rl.question("Enter your choice of grid number = ", (choice) => {
+      return resolve(Number(choice));
+    });
+  });
+};
+
+// TODO: Should I replace store with getStoreChoices()?
+const isValidUserGridChoice = (store, choice) => {
+  if (
+    isNaN(choice) ||
+    choice < 1 ||
+    choice > 9 ||
+    store.choices.includes(choice)
+  ) {
+    console.log(
+      "Please enter a valid grid number. Make sure it hasn't already been picked!"
+    );
+    return false;
+  } else {
+    return true;
+  }
+};
+
+// TODO: Add in a win check somewhere?
+// TODO: Add in grid logging?
```

Comment:
[separation of concerns] This function is very simple, which is good. "win check" and grid logging probably belong in the function that calls this one